sources/background_scripts/notifications.js

```diff
@@ -0,0 +1,25 @@
+NewTabTweaker = {};
+NewTabTweaker.Notifications = {
+	notification_ids: [],
+	notify: function(notification_id, message){
+		var title = browser.runtime.getManifest().name;
+		
+		// Do not display the same message twice by using a notification_id
+		browser.notifications.create(notification_id,
+		{
+			type: "basic",
+			iconUrl: browser.extension.getURL("/icons/main_64.png"),
+			title: title,
+			message: message
+		});
+
+		// Add notificationId to notificationIds, so we can clear it later
+		NewTabTweaker.Notifications.notification_ids.push(notification_id);
+
+		setTimeout(function()
+		{
+			// Clear the first (oldest) notification
+			browser.notifications.clear(NewTabTweaker.Notifications.notification_ids.shift());
+		}, 2200);
```

Comment:
Any reason why this is 2200 and not something more round?

Reply:
When I put it at 2200, I have the time to read the message fully before it appears half a second later. I found 2500 to be too much and 2000 too less. Personal taste, I think.

Reply:
Then maybe the delay should vary with message length? Like set a minimum at something like 2000, and add more for longer messages.

Reply:
^ This may require tweaking the notification clearing code.

Reply:
Otherwise, we should opt for the longer delay (2500), as it is better to have a message stay longer after you read it, than disappear too quickly.